src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/VectorProvidersConfig.java

```diff
@@ -0,0 +1,26 @@
+package io.stargate.sgv2.jsonapi.service.embedding.configuration;
+
+import io.quarkus.runtime.annotations.StaticInitSafe;
+import io.smallrye.config.source.yaml.YamlConfigSource;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Collections;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+@StaticInitSafe
+public class VectorProvidersConfig
+    implements org.eclipse.microprofile.config.spi.ConfigSourceProvider {
+  @Override
+  public Iterable<ConfigSource> getConfigSources(ClassLoader forClassLoader) {
+    try {
+      URL resourceURL = forClassLoader.getResource("vectorProviders.yaml");
+      if (resourceURL != null) {
+        YamlConfigSource configSource = new YamlConfigSource(resourceURL);
+        return Collections.singletonList(configSource);
+      }
```

Comment:
Should also throw an exception if `resourceURL` is `null`?

Reply:
Yes, throw error when resourceURL is null